crates/gossip/src/gossip.rs

```diff
@@ -813,4 +823,253 @@ mod test {
+            )
+            .await
+            .unwrap();
+        harness_1
+            .space
+            .inform_ops_stored(
+                ops.clone().into_iter().map(Into::into).collect(),
+            )
+            .await
+            .unwrap();
+
+        let harness_2 = factory.new_instance().await;
+        harness_2.join_local_agent(DhtArc::FULL).await;
+
+        // Just the first 3 ops should be received, the limit should prevent the next two from
+        // being sent.
+        harness_2
+            .wait_for_ops(
+                ops.iter().take(3).map(|op| op.compute_op_id()).collect(),
+            )
+            .await;
+
+        // We received the ones we wanted above, make sure it's just those 3.
+        let all_ops = harness_2
+            .op_store
+            .retrieve_ops(ops.iter().map(|op| op.compute_op_id()).collect())
+            .await
+            .unwrap();
+        assert_eq!(3, all_ops.len());
+    }
+
+    #[tokio::test]
+    async fn respect_size_limit_for_new_ops_and_dht_disc_diff() {
+        enable_tracing();
+
+        let space = TEST_SPACE_ID;
+        let factory = TestGossipFactory::create(
+            space.clone(),
+            true,
+            Some(K2GossipConfig {
+                max_gossip_op_bytes: 7 * 128,
+                // Set the initiate interval low so that gossip will start quickly after
+                // bootstrapping but leave the min initiate interval high so that we can
+                // run a single gossip round during the test.
+                initiate_interval_ms: 10,
+                ..Default::default()
+            }),
+        )
+        .await;
+        let harness_1 = factory.new_instance().await;
+        let agent_info_1 = harness_1.join_local_agent(DhtArc::FULL).await;
+
+        let mut ops = Vec::new();
+
+        for i in 0u8..5 {
+            let op = MemoryOp::new(
+                Timestamp::from_micros(100 + i as i64),
+                vec![i; 128],
+            );
+            ops.push(op);
+        }
+        harness_1
+            .op_store
+            .process_incoming_ops(
+                ops.clone().into_iter().map(Into::into).collect(),
+            )
+            .await
+            .unwrap();
+
+        let bookmark = Timestamp::now();
+
+        for i in 0u8..5 {
+            let op = MemoryOp::new(Timestamp::now(), vec![100 + i; 128]);
+            ops.push(op);
+        }
+
+        harness_1
+            .op_store
+            .process_incoming_ops(
+                ops.clone().into_iter().skip(5).map(Into::into).collect(),
+            )
+            .await
+            .unwrap();
+        harness_1
+            .space
+            .inform_ops_stored(
+                ops.clone().into_iter().map(Into::into).collect(),
+            )
+            .await
+            .unwrap();
+
+        let harness_2 = factory.new_instance().await;
+
+        // Set a bookmark so that the first 5 ops are behind the bookmark and need a disc sync and
```

Comment:
That implies that new ops take preference over historical ops. Could add that as a comment for people who've not grown up with a gossip algorithm.

Reply:
You mean you didn't have a pet gossip as a child!? :)

I did actually document this, not often I can say that :) Is this okay? https://github.com/holochain/kitsune2/blob/main/crates/gossip/src/config.rs#L11

Reply:
Fair enough!